subsys/bluetooth/host/hci_raw.c

```diff
@@ -71,10 +85,38 @@ struct net_buf *bt_buf_get_evt(u8_t evt, bool discardable, s32_t timeout)
 	return bt_buf_get_rx(BT_BUF_EVT, timeout);
 }
 
+static int bt_h4_recv(struct net_buf *buf)
+{
+	BT_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+
+	switch (bt_buf_get_type(buf)) {
+	case BT_BUF_ACL_IN:
+		net_buf_push_u8(buf, H4_ACL);
+		break;
+	case BT_BUF_EVT:
+		net_buf_push_u8(buf, H4_EVT);
+		break;
+	default:
+		BT_ERR("Unknown type %u", bt_buf_get_type(buf));
+		net_buf_unref(buf);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
 int bt_recv(struct net_buf *buf)
 {
 	BT_DBG("buf %p len %u", buf, buf->len);
 
+	if (IS_ENABLED(CONFIG_BT_HCI_RAW_H4) && raw_mode == BT_RAW_MODE_H4) {
+		int err = bt_h4_recv(buf);
+
+		if (err) {
+			return err;
+		}
```

Comment:
If you're not planning to do anything special with `err` (like logging it) I'd just shorten this to `return bt_h4_recv(buf);`

Reply:
Actually we can't just return bt_h4_recv since that only sets the buffer type so we need to continue to send the buffer to the monitor, etc.